core/src/main/java/com/linecorp/armeria/common/logging/LoggingDecoratorBuilder.java

```diff
@@ -51,13 +53,11 @@
     private LogLevel failedResponseLogLevel = LogLevel.WARN;
```

Comment:
As Trustin said the precedence of `responseLogLevelMapper`, `successfulResponseLogLevel` and `failedResponseLogLevel` are exclusive.
- How about adding a new `ResponseLogLevelMapper` when `successfulResponseLogLevel(LogLevel)` or `failedResponseLogLevel(LogLevel)` is set?
```java
public LoggingDecoratorBuilder successfulResponseLogLevel(LogLevel successfulResponseLogLevel) {
     ResponseLogLevelMapper mapper = log -> log.responseCause() != null ? successfulResponseLogLevel : null;
     // combined the 'mapper' using 'orElse()'
}
``` 
- Should we remove the default values and fields and make a default method using the values?
```java
interface ResponseLogLevelMapper {

    static LogLevel of() {
        return log -> log.responseCause() == null ? LogLevel.DEBUG : LogLevel.WARN;
    }

}
```

Reply:
SGTM 👍 

> Should we remove the default values and fields and make a default method using the values?

How about this?

```java
interface ResponseLogLevelMapper {
    static LogLevel of(LogLevel successfulLogLevel, LogLevel failureLogLevel) {
        return log -> log.responseCause() == null ? successfulLogLevel : failureLogLevel;
    }
}
```

And using `ResponseLogLevelMapper.of(LogLevel.DEBUG, LogLevel.WARN)` when setting default? It may give users more option.

Reply:
> ```
> interface ResponseLogLevelMapper {
>     static LogLevel of(LogLevel successfulLogLevel, LogLevel failureLogLevel) {
>         return log -> log.responseCause() == null ? successfulLogLevel : failureLogLevel;
>     }
> }
> ```

This sounds good to me.